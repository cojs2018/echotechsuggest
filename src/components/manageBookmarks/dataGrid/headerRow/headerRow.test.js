import React from 'react';
import render from 'react-test-renderer';

import HeaderRow from './headerRow';

const mockRows = [
    {field: 'bookmarkId', headerName: 'ID', width: 10},
    {field: 'articleName', headerName: 'Article Name', width: 20},
    {field: 'path', headerName: 'Path', width: 50},
    {field: 'url', headerName: 'URL', width: 50},
    {field: 'createdAt', headerName: 'Created At', width: 20},
    {field: 'tags', headerName: 'Tags', width: 100}
]

describe('HeaderRows', () => {
    it('renders correctly', () => {
        const props = {
            columns: mockRows
        }

        const tree = render
            .create(<HeaderRow {...props} />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    })
})