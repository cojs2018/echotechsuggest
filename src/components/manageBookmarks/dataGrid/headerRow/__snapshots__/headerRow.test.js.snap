// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderRows renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "height": 48,
        "paddingHorizontal": 16,
      },
      Object {
        "borderBottomColor": "rgba(0, 0, 0, 0.12)",
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignContent": "center",
          "flex": 1,
          "flexDirection": "row",
          "paddingVertical": 12,
        },
        undefined,
        undefined,
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          Array [
            Object {
              "alignItems": "center",
              "fontSize": 12,
              "fontWeight": "500",
              "height": 24,
              "lineHeight": 24,
            },
            Object {
              "color": "rgba(0, 0, 0, 0.6)",
            },
          ],
        ]
      }
    >
       
    </Text>
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignContent": "center",
          "flex": 1,
          "flexDirection": "row",
          "paddingVertical": 12,
        },
        undefined,
        Object {
          "width": 10,
        },
      ]
    }
    testID="bookmarkId"
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          Array [
            Object {
              "alignItems": "center",
              "fontSize": 12,
              "fontWeight": "500",
              "height": 24,
              "lineHeight": 24,
            },
            Object {
              "color": "rgba(0, 0, 0, 0.6)",
            },
          ],
        ]
      }
    >
      ID
    </Text>
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignContent": "center",
          "flex": 1,
          "flexDirection": "row",
          "paddingVertical": 12,
        },
        undefined,
        Object {
          "width": 20,
        },
      ]
    }
    testID="articleName"
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          Array [
            Object {
              "alignItems": "center",
              "fontSize": 12,
              "fontWeight": "500",
              "height": 24,
              "lineHeight": 24,
            },
            Object {
              "color": "rgba(0, 0, 0, 0.6)",
            },
          ],
        ]
      }
    >
      Article Name
    </Text>
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignContent": "center",
          "flex": 1,
          "flexDirection": "row",
          "paddingVertical": 12,
        },
        undefined,
        Object {
          "width": 50,
        },
      ]
    }
    testID="path"
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          Array [
            Object {
              "alignItems": "center",
              "fontSize": 12,
              "fontWeight": "500",
              "height": 24,
              "lineHeight": 24,
            },
            Object {
              "color": "rgba(0, 0, 0, 0.6)",
            },
          ],
        ]
      }
    >
      Path
    </Text>
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignContent": "center",
          "flex": 1,
          "flexDirection": "row",
          "paddingVertical": 12,
        },
        undefined,
        Object {
          "width": 50,
        },
      ]
    }
    testID="url"
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          Array [
            Object {
              "alignItems": "center",
              "fontSize": 12,
              "fontWeight": "500",
              "height": 24,
              "lineHeight": 24,
            },
            Object {
              "color": "rgba(0, 0, 0, 0.6)",
            },
          ],
        ]
      }
    >
      URL
    </Text>
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignContent": "center",
          "flex": 1,
          "flexDirection": "row",
          "paddingVertical": 12,
        },
        undefined,
        Object {
          "width": 20,
        },
      ]
    }
    testID="createdAt"
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          Array [
            Object {
              "alignItems": "center",
              "fontSize": 12,
              "fontWeight": "500",
              "height": 24,
              "lineHeight": 24,
            },
            Object {
              "color": "rgba(0, 0, 0, 0.6)",
            },
          ],
        ]
      }
    >
      Created At
    </Text>
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignContent": "center",
          "flex": 1,
          "flexDirection": "row",
          "paddingVertical": 12,
        },
        undefined,
        Object {
          "width": 100,
        },
      ]
    }
    testID="tags"
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          Array [
            Object {
              "alignItems": "center",
              "fontSize": 12,
              "fontWeight": "500",
              "height": 24,
              "lineHeight": 24,
            },
            Object {
              "color": "rgba(0, 0, 0, 0.6)",
            },
          ],
        ]
      }
    >
      Tags
    </Text>
  </View>
</View>
`;
